app/views/exams/ProgressPage.vue

```diff
@@ -0,0 +1,128 @@
+<template>
+  <div class="progress-page center-div">
+    <div class="header">
+      <div class="code-language">
+        {{ userExam.codeLanguage }}
+      </div>
+      <div class="timer">
+        Time Left: {{ timeLeft }}
+      </div>
+    </div>
+    <div class="levels">
+      <exam-level
+        v-for="(level, index) in problems"
+        :key="level._id"
+        :level="level"
+        :language="userExam.codeLanguage"
+        :index="index + 1"
+      />
+    </div>
+
+    <div class="submit center-div">
+      <input
+        type="button"
+        value="Mark as Complete"
+        @click="submit"
+      >
+      <div class="tip">
+        Are you sure you want to mark as complete? You will not be able to play the levels again.
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters, mapActions } from 'vuex'
+import ExamLevel from './components/ExamLevel'
+export default {
+  components: {
+    ExamLevel,
+  },
+  props: {
+    examId: {
+      type: String,
+      required: true,
+    },
+  },
+  data () {
+    return {
+      timeLeft: '00:00',
+      counterInterval: null,
+    }
+  },
+  computed: {
+    ...mapGetters('exams', [
+      'getExamById',
+      'userExam',
+    ]),
+    exam () {
+      return this.getExamById(this.examId)
+    },
+    problems () {
+      if (!this.exam) {
+        return []
+      }
+      const problems = this.exam.problems
+      const levels = []
+      problems.forEach((courseLevels) => {
+        courseLevels.levels.forEach((level, index) => {
+          levels.push({
+            ...level,
+            courseId: courseLevels.courseId,
+            instanceId: courseLevels.instanceId,
+          })
+        })
+      })
+      return levels
+    },
+  },
+  mounted () {
+    this.counter()
+    const oneMin = 60 * 1000
+    this.counterInterval = setInterval(this.counter, oneMin)
+  },
+  beforeDestroy () {
+    clearInterval(this.counterInterval)
+  },
+  methods: {
+    ...mapActions('exams', [
+      'submitExam',
+    ]),
+    paddingZero (num) {
+      return `00${num}`.slice(-2)
+    },
+    counter () {
+      const oneMin = 60 * 1000
+      const startDate = new Date(this.exam.startDate)
+      const minsElapse = (new Date() - startDate) / oneMin
+      let minsLeft = this.exam.duration - minsElapse
+      if (minsLeft <= 0) {
```

Comment:
some logic issue here not letting be on this page.
Console.log response in screenshot for this code:
```
let minsLeft = this.exam.duration - minsElapse
console.log('counter', minsLeft, minsElapse, this.exam.duration) 
```
<img width="374" alt="Screenshot 2024-10-21 at 1 14 32 PM" src="https://github.com/user-attachments/assets/00d8da56-f2fd-4e80-89cf-75c60b793f53">